src/miral/basic_window_manager.cpp

```diff
@@ -1592,6 +1592,7 @@ auto miral::BasicWindowManager::select_active_window(Window const& hint) -> mira
     {
         if (!info_for_hint.is_visible())
         {
+            update_attached_and_fullscreen_sets(info_for_hint, mir_window_state_restored);
```

Comment:
This simply looks wrong: we're navigating a tree to find the window to focus on, so why start updating the attachment sets? Based on a faked state??

Reply:
Oh, looking at the context I see what is happening.

The logic around there is wrong. Just because an invisible window can be activated and isn't _hidden_ doesn't mean it can be restored!